src/tm-threads.c

```diff
@@ -1713,16 +1766,32 @@ TmEcode TmThreadSpawn(ThreadVars *tv)
         return TM_ECODE_FAILED;
     }
 
-    /* Initialize and set thread detached attribute */
-    pthread_attr_init(&attr);
+#ifdef HAVE_DPDK
+    /* only DPDK worker threads are launched using rte_eal_remote_launch */
+    if (tv->lcore_id != LCORE_ID_ANY) {
+        struct dpdk_callback cb = { .func = tv->tm_func, .tv = tv };
 
-    pthread_attr_setdetachstate(&attr, PTHREAD_CREATE_JOINABLE);
+        int rc = rte_eal_remote_launch(&TmThreadSpawnDpdkThread, (void *)&cb, tv->lcore_id);
```

Comment:
@lukashino as recommended there is no requirement to use rte_remote_launch and one should use the worker core id from suricata populated data . please rework

Reply:
For future compatibility, I would rather stick with the launch of DPDK lcores. Why should I run pthreads with limited functionality of DPDK libraries? 
Additionally, I can not use worker core id (I think you meant field `tv->cpu_affinity` and the function `AffinityGetNextCPU`) because at the time of thread launch the information is not assigned yet. 
But as mentioned before, the thread can be later repinned according to the configuration in the suricata.yaml file.